components/puppeteer/actions/common/common.mjs

```diff
@@ -0,0 +1,19 @@
+export default {
+  props: {
+    url: {
+      type: "string",
+      label: "URL",
+      description:
+        "The URL of the page to scrape. For example, `https://example.com`.",
+    },
+  },
+  methods: {
+    async run({ $ }) {
```

Comment:
I think this can be a synchronous method and named something else to avoid confusion, also $ is not used at all so it doesn't have to be passed in
```suggestion
    getFilteredUrl() {
```